packages/-ember-data/addon/setup-container.js

```diff
@@ -1,62 +1,54 @@
+        `This test relies on a deprecated test setup that is no longer supported by EmberData.` +
+        ` To resolve this you will need to be on a recent version of @ember/test-helpers` +
+        ` AND your tests must use \`setApplication()\` instead of \`setResolver()\` and` +
+        ` \`module()\` with \`setup*Test()\`instead of \`moduleFor*()\`.`,
+      false,
+      {
+        id: 'ember-data:legacy-test-helper-support',
+        until: '3.17',
+      }
+    );
+
+    application.optionsForType('serializer', { singleton: false });
+    application.optionsForType('adapter', { singleton: false });
+
+    if (!application.has('service:store')) {
+      application.register('service:store', Store);
+    }
+
+    return;
   }
-  return application.hasRegistration(registrationName);
-}
-/*
- Configures a registry for use with an Ember-Data
- store. Accepts an optional namespace argument.
 
- @method initializeStore
- @param {Ember.Registry} registry
- */
-function initializeStore(application) {
-  // we can just use registerOptionsForType when setupStore is killed
-  // see https://github.com/emberjs/data/issues/6357
-  let registerOptionsForType = application.registerOptionsForType || application.optionsForType;
-  registerOptionsForType.call(application, 'serializer', { singleton: false });
-  registerOptionsForType.call(application, 'adapter', { singleton: false });
+  application.registerOptionsForType('serializer', { singleton: false });
+  application.registerOptionsForType('adapter', { singleton: false });
 
-  if (!hasRegistration(application, 'service:store')) {
+  if (!application.hasRegistration('service:store')) {
     application.register('service:store', Store);
   }
 }
 
-/*
- Configures a registry with injections on Ember applications
- for the Ember-Data store. Accepts an optional namespace argument.
-
- @method initializeDebugAdapter
- @param {Ember.Registry} registry
- */
-function initializeDataAdapter(registry) {
-  registry.register('data-adapter:main', DebugAdapter);
+function initializeDataAdapter(application) {
+  application.register('data-adapter:main', DebugAdapter);
 }
 
-/*
- Configures a registry with injections on Ember applications
- for the Ember-Data store. Accepts an optional namespace argument.
-
- @method initializeStoreInjections
- @param {Ember.Registry} registry
- */
-function initializeStoreInjections(registry) {
-  // registry.injection for Ember < 2.1.0
-  // application.inject for Ember 2.1.0+
-  let inject = registry.inject || registry.injection;
-  inject.call(registry, 'controller', 'store', 'service:store');
-  inject.call(registry, 'route', 'store', 'service:store');
-  inject.call(registry, 'data-adapter', 'store', 'service:store');
+function initializeStoreInjections(application) {
+  application.inject('controller', 'store', 'service:store');
+  application.inject('route', 'store', 'service:store');
+  application.inject('data-adapter', 'store', 'service:store');
```

Comment:
Looks like, based on the external partner tests, that we may need to add a similar deprecation for `registry.injection` or keep it around...What are your thoughts @runspired?

Reply:
The one deprecation is enough but we need to restore the deprecated code path